src/components/Posts/IndividualPost/IndividualPost.js

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from 'react';
+import Typography from '@material-ui/core/Typography';
+import Container from '@material-ui/core/Container';
+import Chip from '@material-ui/core/Chip';
+import cookie from 'react-cookies';
+import { serviceRequest } from '../../../services/serviceRequest';
+import { ErrorMessage } from '../../ErrorMessage';
+import useStyles from './style';
+import config from '../../../config';
+
+const path = '/api/v1/user/getpostbyid?';
+const domain = config.apiDomain;
+
+const preparePayload = (method, headers, params) => {
+  const url = `https://${domain}${path}`;
+  return {
+    method,
+    url,
+    headers,
+    params,
+  };
+};
+
+export const IndividualPost = () => {
+  const classes = useStyles();
+  const [post, setPost] = useState({
+    label: 'sample',
+    title: 'sample title',
+    content: 'This is a sample content for the sample post with sample title',
+    updatedAt: '2019-11-07T09:26:51.822Z',
+    username: 'sampleMe',
+    postID: 1,
+    posterID: 1,
+  });
+  const [errorInfo, setErrorInfo] = useState({ isError: false, errorMsg: null });
+
+  useEffect(() => {
+    const urls = window.location.href;
+    const postId = urls.slice(urls.lastIndexOf('/') + 1, urls.length);
+    const token = cookie.load('jwtToken');
+    const postHeaders = { Authorization: token };
+
+    const fetchPost = async () => {
+      try {
+        const requestPayload = preparePayload('get', postHeaders,
+          { postID: postId });
+        const response = await serviceRequest(requestPayload);
+        const isValidPost = response.status && response.status === 'success';
+        if (isValidPost) {
```

Comment:
Could be written in one line inside if statement but is a matter of style

Reply:
this is an infinite discussion, song and you provide different different coding style suggestion~